### DISCUSSION TITLE
InvalidDatasourceError: Datasource URL must use prisma:// protocol when --data-proxy is used

### DISCUSSION THREAD

**Discussion by elie222** [prisma/prisma#14387] [Q&A] [answered]
I keep getting this error:
"InvalidDatasourceError: Datasource URL must use prisma:// protocol when --data-proxy is used"
when trying to use Prisma Data Proxy.

I have a `prisma://` url set. Using Prisma 4.1.0

Seems like a bug in Prisma. Trying to get it working with Next.js when running locally.

I used this command to generate:

```
PRISMA_GENERATE_DATAPROXY=true npx prisma generate --data-proxy
```

**Comment by elie222** [comment, marked as answer]
Sorry. Found my answer. I console logged the env var and it was indeed set wrong. Using `datasources: { url }` didn't work though. I guess it checked the env and crashed before even getting there

**Reply by janpio** [reply]
`datasources` overrides do not work with Data Proxy Client yet: https://github.com/prisma/prisma/issues/11595 Will be fixed soon hopefully.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
